zerver/tests/test_events.py

```diff
@@ -3574,6 +3578,155 @@ def do_set_realm_permission_group_setting_test(self, setting_name: str) -> None:
 
             old_group_id = new_group_id
 
+    def do_set_realm_permission_group_setting_to_anonymous_groups_test(
+        self, setting_name: str
+    ) -> None:
+        realm = self.user_profile.realm
+        system_user_groups_dict = get_role_based_system_groups_dict(
+            realm,
+        )
+
+        setting_permission_configuration = Realm.REALM_PERMISSION_GROUP_SETTINGS[setting_name]
+
+        default_group_name = setting_permission_configuration.default_group_name
+        default_group = system_user_groups_dict[default_group_name]
+
+        now = timezone_now()
+
+        do_change_realm_permission_group_setting(
+            realm,
+            setting_name,
+            default_group,
+            old_setting_api_value=getattr(realm, setting_name).id,
+            acting_user=self.user_profile,
+        )
```

Comment:
It might be better to just make all the changes via the API, to avoid having to do the `old_setting_api_value=getattr(realm, setting_name).id` hackery.

Reply:
We consistently use do_... functions in the event tests. I have anyways made the old_setting_api_value optional field.

Reply:
Yeah, that's a fine solution.